### ISSUE TITLE
Unable to use portaudio::portaudio via. Cmake as usual practices in my project

### ISSUE THREAD

**Issue by unbadfish** [PortAudio/portaudio#748]
Hello, Portaudio team.
I tried to install portaudio as turtorials in http://files.portaudio.com/docs/v19-doxydocs/compile_cmake.html
and I have done these things to install

1. make a empty folder `livepitch`
2. download the portaudio-19.7.0 stable in https://github.com/PortAudio/portaudio/releases/tag/v19.7.0
3. exact it and copy to `livepitch/resource`
4. edit CMakeLists.txt as https://github.com/PortAudio/portaudio/issues/486#issuecomment-769053675 mentioned
5. mkdir `livepitch/include`
6. cd livepitch/include
7. run `cmake d:/Projects/livepitch/resource/portaudio-19.7.0  -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX=d:/Projects/livepitch/include`
8. run `make`
9. run `make install`

After the installation, I create  `CMakeLists.txt` and `main.c`, and the contents are below:
```cmake
#CMakeLists.txt
cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
project(test1)
set(CMAKE_MODULE_PATH_PATH ${CMAKE_SOURCE_DIR}/include)
find_package(portaudio)

if(NOT portaudio_FOUND)
    message(FATAL_ERROR "PortAudio Not Found!")
endif(NOT portaudio_FOUND)

add_executable(mainexe main.c)

target_link_libraries(mainexe portaudio::portaudio)
                            # As usual practice
```
```c
//main.c
#include <stdio.h>
#include "func.h"
#include "portaudio.h"
#include "pa_util.h"
int main(void){
    int version = 0;
    version = Pa_GetVersion();
    printf("%d", version);
    return 0;}
```
Then, I cd to livepitch/build , run `cmake .. -G "Unix Makefiles"`
And an error occured:
```
$ cmake .. -G "Unix Makefiles"
-- Configuring done
CMake Error at CMakeLists.txt:12 (target_link_libraries):
  Target "mainexe" links to:

    portaudio::portaudio

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
As the `if(NOT portaudio_FOUND)` was not activated, I assume that the portaudio is imported, but I cannot use it.

What I also tried: 
Change `target_link_libraries(mainexe portaudio::portaudio)` to `target_link_libraries(mainexe portaudio)`
the cmake command didn't cause error, but when I run `make`, error occured:
```
Lenovo@DESKTOP-------- MINGW64 /d/Projects/livepitch/build
$ cmake .. -G "Unix Makefiles"
-- Configuring done
-- Generating done
-- Build files have been written to: D:/Projects/livepitch/build

Lenovo@DESKTOP-------- MINGW64 /d/Projects/livepitch/build
$ make
[ 50%] Building C object CMakeFiles/mainexe.dir/main.c.obj
D:/Projects/livepitch/main.c:4:10: fatal error: pa_util.h: No such file or directory
    4 | #include "pa_util.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/mainexe.dir/build.make:76：CMakeFiles/mainexe.dir/main.c.obj] 2 ERROR(s)
make[1]: *** [CMakeFiles/Makefile2:83：CMakeFiles/mainexe.dir/all] 2 ERROR(s)
make: *** [Makefile:91：all] 2 ERROR(s)
```
My executable cannot include `pa_util.h`(and everything else in `src/`)

So, PLEASE do the following things to improve development efficiency:

- provide a template for cmake developing (contains  main.c, CMakeLists.txt, func.c, func.f；all .c source files include on or more portaudio source file)
- record a detailed video to tell developers how to build and use portaudio in all envirments(e.g. cmake, VS)
- update the document website

My OS / toolchain
 - OS: Win 11
 - PortAudio version: portaudio-19.7.0 stable in https://github.com/PortAudio/portaudio/releases/tag/v19.7.0
 - If Windows or Linux, which Host API : ASIOSDK (I placed it in `d:/Projects/livepitch/resource/portaudio-19.7.0/src/hostapi/ASIOSDK`, and I decided to move it to `d:/Projects/livepitch/include` after the major issue was fixed)
 - MSYS2 MinGW x86
 - cmake version 3.24.2
 - gcc.exe (Rev1, Built by MSYS2 project) 12.2.0

Here is the tree of my project
```
├─build
│  ├─.cmake
│  │  └─api
│  │      └─v1
│  │          ├─query
│  │          │  └─client-vscode
│  │          └─reply
│  └─CMakeFiles
│      ├─3.24.2
│      │  ├─CompilerIdC
│      │  │  └─tmp
│      │  └─CompilerIdCXX
│      │      └─tmp
│      ├─CMakeTmp
│      ├─mainexe.dir
│      ├─pkgRedirects
│      └─Progress
├─include
│  ├─bin
│  ├─cmake
│  │  └─portaudio
│  ├─CMakeFiles
│  │  ├─3.24.2
│  │  │  ├─CompilerIdC
│  │  │  │  └─tmp
│  │  │  └─CompilerIdCXX
│  │  │      └─tmp
│  │  ├─CMakeTmp
│  │  ├─Export
│  │  │  └─90a8b282bce0779409342a760050f91f
│  │  ├─pkgRedirects
│  │  ├─portaudio.dir
│  │  │  └─src
│  │  │      ├─common
│  │  │      ├─hostapi
│  │  │      │  ├─dsound
│  │  │      │  ├─skeleton
│  │  │      │  ├─wasapi
│  │  │      │  ├─wdmks
│  │  │      │  └─wmme
│  │  │      └─os
│  │  │          └─win
│  │  ├─portaudio_static.dir
│  │  │  └─src
│  │  │      ├─common
│  │  │      ├─hostapi
│  │  │      │  ├─dsound
│  │  │      │  ├─skeleton
│  │  │      │  ├─wasapi
│  │  │      │  ├─wdmks
│  │  │      │  └─wmme
│  │  │      └─os
│  │  │          └─win
│  │  └─uninstall.dir
│  ├─include
│  ├─lib
│  │  ├─cmake
│  │  │  └─portaudio
│  │  └─pkgconfig
│  └─share
│      └─doc
│          └─portaudio
├─resource
│  ├─portaudio-19.7.0【content skipped】
│
├─CMakeLists.txt
├─main.c
```

**Comment by RossBencina** [comment]
There's already a ticket for fixing the documentation. I agree that it needs to be fixed.

https://github.com/PortAudio/portaudio/issues/662

**Comment by unbadfish** [comment]
I have double-checked my code and I found that:

1. Only the (.h)files in`DCMAKE_INSTALL_PREFIX/include`(after install) can be included using `#include "___.h"`, they are :` pa_asio.h  pa_win_ds.h  pa_win_wasapi.h  pa_win_waveformat.h  pa_win_wdmks.h  pa_win_wmme.h  portaudio.h`.
2. Other (.h)files in `portaudio-19.7.0/src/common` can not be included using `#include "___.h".
3. When defined`-DPA_BUILD_EXAMPLES=ON  -DPA_BUILD_TESTS=ON`, the example/test code (which **includes header files not in `DCMAKE_INSTALL_PREFIX/include`**) can include the header files and source files and can be linked corrcetly.
4. All files in https://github.com/PortAudio/portaudio/blob/f217b421cf7336a5e7cc03181289df6c8974966e/CMakeLists.txt#L12-L37 can not be included, while I wrote `find_package(PortAudio)` and `target_link_libraries(mainexe PortAudio)` in CMakeList.txt

I suggest that : When installing, print a message that what branch in cmakelist they have went into (for example:WIN32 and PA_USE_ASIO),  wrote the message hard-code to the program, and add a function to show the message.

**Comment by RossBencina** [comment]
Files in 'src/common' are private to PortAudio implementation and definitely should *not* be includable when PortAudio is installed.

**Comment by unbadfish** [comment]
Thanks! I have changed my code 🤝 . But I still have some question:
In here : https://github.com/PortAudio/portaudio/blob/f217b421cf7336a5e7cc03181289df6c8974966e/examples/paex_ocean_shore.c#L67
The example code still use `#include "pa_util.h"` , and it is not the only file that use these "no-includable" files 😢, the example code is not leading the user to the right way
So is it possible that we can solve the problem like this

```c
#ifdef __asio__
#define xxx _xxxasio
#endif __asio__
```
so whatever the platform is, the API appears the same way, and we can use the same implementation api without problem

**Comment by RossBencina** [comment]
`paex_ocean_shore.c` is non-conforming and needs to be fixed or removed. there is already a ticket for that: #226

**Comment by unbadfish** [comment]
Fixed by #815 .
😄
